### DISCUSSION TITLE
Security Issue With Pimcore Login

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#13284] [Q&A / Support]
Hi All,
@
_**We use the only Pimcore Admin for the application and the application is running on SSL**_

We have scanned the Pimcore application into the [Qualys](https://success.qualys.com/) tool and we found the below security issue in the report.

 -->> 150052 Password is present in HTTP traffic unrelated to the login page 

**Threat**
The password used for an authenticated scan was present in a request or response not directly related to the authentication process (i.e. in a location other than 
the login page). This demonstrates poor password handling by the web application, which should never expose a user's authentication credentials.

**Impact**
The web application exposes a user's password outside of the login process. A user's password should never be reused, reflected, or otherwise present in web 
traffic except during the authentication process.

**Solution**
The password should only be transferred during an authentication process. It should be stored by the web application in a hashed format using a strong hashing 
algorithm that includes a salt or a mechanism like PBKDF2. A strong pseudo-random token should be used in place of the password or its hash if the value must 
be transferred during a transaction other than authentication. All traffic that involves an authenticated user should use HTTPS or HTTP Strict Transport Security (HSTS)

In this report, they share a solution also but I don't know how to fix it on the Pimcore admin login.

Also, I have checked the Pimcore login request and in this request, we can see the username & password in plain text.
![image](https://user-images.githubusercontent.com/52573912/193259838-6d178ba5-ec25-4035-a951-3548bea93b1e.png)
  
Anyone can guide how we can fix it? 

What we have tried is to overwrite the admin login and decode the password, so right now whatever password is posted in a request that was encoded, but still, this issue is not resolved.

**Comment by NiklasBr** [comment]
Looks like a false positive to me.

You are on `/admin` and POST:ing the credentials to `/admin/login/login` which is expected and nothing unusual by having a login form not exactly matching "login" API endpoint. Either that or your tool by some means sends it twice?

Attempting to find the credentials in any other POST data on the demo site fails for me, so it is only sent once.

**Reply by ghost** [reply]
Thanks, @NiklasBr  for the quick replay.

**Comment by fashxp** [comment]
just for the future: please follow our [policy](https://github.com/pimcore/pimcore/security/policy) for security topics and do not expose them in the discussions or issues!
